src/Middleware/src/Headstart.API/Commands/SupplierSync/EntityCommands/GenericSupplierCommand.cs

```diff
@@ -32,16 +34,28 @@ public GenericSupplierCommand(AppSettings settings)
             });
         }
 
-        public async Task<JObject> GetOrderAsync(string ID, DecodedToken decodedToken)
+        public async Task<JObject> GetOrderAsync(string ID, OrderType orderType, DecodedToken decodedToken)
         {
             //TODO: BaseUrl cannot be found here
             var ocAuth = await _ocSeller.AuthenticateAsync();
             HSShipEstimate estimate;
             HSShipMethod ship_method = null;
-            var supplierWorksheet = await _ocSeller.IntegrationEvents.GetWorksheetAsync<HSOrderWorksheet>(OrderDirection.Outgoing, ID, ocAuth.AccessToken);
-            
-            var buyerWorksheet = await _ocSeller.IntegrationEvents.GetWorksheetAsync<HSOrderWorksheet>(OrderDirection.Incoming, ID.Split('-')[0], ocAuth.AccessToken);
-            var buyerLineItems = buyerWorksheet.GetBuyerLineItemsBySupplierID(supplierWorksheet.Order.ToCompanyID);
+            HSOrderWorksheet supplierWorksheet = null;
+
+            // Supplier worksheet will not exist on quote orders.
+            try
+            {
+                supplierWorksheet = await _ocSeller.IntegrationEvents.GetWorksheetAsync<HSOrderWorksheet>(OrderDirection.Outgoing, ID, ocAuth.AccessToken);
+            }
+            catch (OrderCloudException) {}
+            var salesOrderID = orderType == OrderType.Standard ? ID.Split('-')[0] : ID;
+            var buyerWorksheet = await _ocSeller.IntegrationEvents.GetWorksheetAsync<HSOrderWorksheet>(OrderDirection.Incoming, salesOrderID, ocAuth.AccessToken);
+            var supplierID = supplierWorksheet?.Order?.ToCompanyID;
+            if (buyerWorksheet.Order.xp?.OrderType == OrderType.Quote)
+            {
+                supplierID = buyerWorksheet.LineItems[0].SupplierID;
+            }
+            var buyerLineItems = buyerWorksheet.GetBuyerLineItemsBySupplierID(supplierID);
```

Comment:
Does this need to be modified if supplierID is null?

Reply:
I don't think so - if the supplier ID is null (admin-owned products) it would just select the admin lines at that stage, which would be expected.

Reply:
Got it makes sense